cmd/step-pkcs11-init/main.go

```diff
@@ -297,6 +315,12 @@ func checkObject(k kms.KeyManager, rawuri string) {
 	}
 }
 
+var ktyMapping = map[string]apiv1.SignatureAlgorithm{
+	"EC":  apiv1.ECDSAWithSHA256,
+	"RSA": apiv1.SHA256WithRSAPSS,
```

Comment:
Some PKCS#11 modules like YubiHSM2 can sign using `RSA-PSS`, but the same key can sign using `RSA-PKCS#1 v1.5` and it will do that by default. For example, running this code as it is will sign both certificates using PKCS#1 instead of PSS. If you want to use PSS you need to add to the templates `SignatureAlgorithm: x509.SHA256WithRSAPSS`. 

Then as YubiHSM2 can sign with both, if you want to force RSA-PSS you will need to set templates on each provisioner and manually specify the signature algorithm too.

The behavior is different for other KMSs, for example, in Google CloudKMS, if you create a key for SHA256+RSA-PSS signing, it can only sign with it, `step-ca` is able to detect this and sets the appropriate SignatureAlgorithm.

It might make sense for other KMSs to inherit the signature algorithm from the certificate if an RSA key is used, but right now it uses Go's default, in this case, RSA PKCS#1.

In summary, right now you have a couple of options, change this to be `apiv1.SHA256WithRSA` or add `SignatureAlgorithm: x509.SHA256WithRSAPSS` to the templates in this file and be aware of the known issues.

Reply:
Related https://github.com/smallstep/certificates/issues/952